### ISSUE TITLE
How to test using maven

### ISSUE THREAD

**Issue by tcho16** [bluebird75/luaunit#86]
Hi I've currently written tests using the lua busted framework but turns out there isn't any maven plugins for busted. I came across luaunit and you claim that I can use luaunit in maven. I tried looking for the luaunit maven plugin but couldn't find any. Could you tell me how I could incorporate this luaunit test on maven. Thank you.

**Comment by n1tehawk** [comment]
Hi tcho16!

**DISCLAIMER: I have no personal experience with configuring/running Maven at all.**

First of all, it's important to realize that neither _LuaUnit_ nor _busted_ offer direct out-of-the-box support for Maven (i.e. provide a suitable plugin).

Instead, they both allow output of test results as XML in a JUnit-compatible format ("xUnit"). For `LuaUnit` you would use options like `--output junit --name myresults.xml`, and a quick check shows that _busted_ has a similar `outputHandler`.

If I understand it correctly, Maven integration thus consists of two steps:
1. Execute the (Lua) script that does the actual testing using _LuaUnit_ (or _busted_), and creates corresponding JUnit output.
2. Make the test results available/meaningful to Maven by parsing the XML resulting from step 1.

Questions related to step 1. have been asked several times on StackOverflow - [here's an answer](http://stackoverflow.com/questions/3491937/i-want-to-execute-shell-commands-from-mavens-pom-xml) that I found useful. If you are content with a simple `BUILD SUCCESSFUL` vs. `BUILD FAILED` result, this might already be all that is required.

Step 2. seems to be the more interesting one. While searching for a solution, I came across https://github.com/qqilihq/maven-test-parser-plugin which seems very helpful.

Regards, NiteHawk
